src/systems/physics-system.js

```diff
@@ -0,0 +1,101 @@
+import { World } from "three-ammo";
+import { TYPE } from "three-ammo/constants";
+
+const WORLD_CONFIG = {
+  debugDrawMode: THREE.AmmoDebugConstants.DrawWireframe,
+  gravity: { x: 0, y: -9.8, z: 0 }
+};
+
+export class PhysicsSystem {
+  constructor(scene) {
+    const Ammo = require("ammo.js/builds/ammo.wasm.js");
+    const AmmoWasm = require("ammo.js/builds/ammo.wasm.wasm");
+    const AmmoModule = Ammo.bind(undefined, {
+      locateFile(path) {
+        if (path.endsWith(".wasm")) {
+          return AmmoWasm;
+        }
+        return path;
+      }
+    });
+
+    this.bodies = [];
+    this.bodyHelpers = [];
+    this.shapeHelpers = [];
+
+    this.debugRequested = false;
+    this.debugEnabled = false;
+    this.scene = scene;
+    this.stepDuration = 0;
+
+    AmmoModule().then(() => {
+      this.world = new World(WORLD_CONFIG);
+    });
+  }
+
+  setDebug(debug) {
+    this.debugRequested = debug;
+  }
+
+  tick(dt) {
+    if (this.world) {
+      if (this.debugRequested !== this.debugEnabled) {
+        this.debugEnabled = this.debugRequested;
+        if (this.debugEnabled) {
+          this.world.getDebugDrawer(this.scene).enable();
+        } else {
+          this.world.getDebugDrawer(this.scene).disable();
+        }
+      }
```

Comment:
Why did you choose to do this in `tick`, instead of having `setDebug` call functions on the debug drawer directly?

Reply:
This is due to a possible race condition where if you use the `?debug=true` flag, `setDebug` can get called before `world` exists.

Reply:
I see, that makes sense. I might have implemented `setDebug` in terms of a loading promise instead, such as
```javascript
setDebug: function(enabled) {
  this.whenReady().then(world => {
    world.getDebugDrawer(this.scene).enabled = enabled;
  });
}
```
But it's a matter of taste.